src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        Animal.removeDeadAnimals();
+      }
+    } else if (!(target instanceof Carnivore)) {
+      return 'Invalid target!';
+    }
```

Comment:
The condition `else if (!(target instanceof Carnivore))` is misleading. It implies that a `Carnivore` can only bite a `Herbivore` or another `Carnivore`, but the logic should only allow biting `Herbivore` instances. Consider changing this condition to `else` to handle any invalid target that is not a `Herbivore`.